zwave_js_server/__main__.py

```diff
@@ -74,31 +77,38 @@ async def handle_dump_state(
 async def connect(args: argparse.Namespace, session: aiohttp.ClientSession) -> None:
     """Connect to the server."""
     client = Client(args.url, session)
-    asyncio.create_task(client.connect())
+    asyncio.create_task(register_controller_listeners(client))
 
-    def log_value_updated(event: dict) -> None:
-        node = event["node"]
-        value = event["value"]
+    async with client:
+        await client.listen()
+
+
+def log_value_updated(event: dict) -> None:
+    """Log node value changes."""
+    node = event["node"]
+    value = event["value"]
+
+    if node.device_config:
+        description = node.device_config.description
+    else:
+        description = f"{node.device_class.generic} (missing device config)"
+
+    logger.info(
+        "Node %s %s (%s) changed to %s",
+        description,
+        value.property_name or "",
+        value.value_id,
+        value.value,
+    )
 
-        if node.device_config:
-            description = node.device_config.description
-        else:
-            description = f"{node.device_class.generic} (missing device config)"
-
-        logger.info(
-            "Node %s %s (%s) changed to %s",
-            description,
-            value.property_name or "",
-            value.value_id,
-            value.value,
-        )
 
+async def register_controller_listeners(client: Client) -> None:
```

Comment:
This can just be inlined in the connect function. We no longer need to wait for `initialized` event or run it as a new task.

Reply:
In fact, we could run into a race condition right now where initialized is achieved before the listener is attached.